typing/typecore.ml

```diff
@@ -914,18 +912,18 @@ and build_as_type_aux ~refine (env : Env.t ref) p =
       let mode =
         match Ctype.immediacy !env p.pat_type with
         | Always -> Value_mode.newvar ()
-        | Unknown | Always_on_64bits -> p.pat_mode
+        | Unknown | Always_on_64bits -> mode
       in
       p.pat_type, mode
   | Tpat_any | Tpat_var _
   | Tpat_array _ | Tpat_lazy _ ->
-      p.pat_type, p.pat_mode
+      p.pat_type, mode
 
 (* Constraint solving during typing of patterns *)
 
-let solve_Ppat_alias ~refine env pat =
+let solve_Ppat_alias ~refine ~mode env pat =
   begin_def ();
-  let ty_var, mode = build_as_type_and_mode ~refine env pat in
+  let ty_var, mode = build_as_type_and_mode ~refine ~mode env pat in
   end_def ();
   generalize ty_var;
   ty_var, mode
```

Comment:
maybe this function does not need to return `mode` anymore, just to simplify the code a bit?

Reply:
my bad - the mode could be updated.